x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -1225,6 +1307,231 @@ export class RulesClient {
     );
   }
 
+  public async bulkEdit<Params extends RuleTypeParams>({
+    filter,
+    actions,
+    paramsModifier,
+  }: BulkEditOptions<Params>): Promise<{
+    rules: Array<SanitizedRule<Params>>;
+    errors: BulkEditError[];
+  }> {
+    let authorizationTuple;
+    try {
+      authorizationTuple = await this.authorization.getFindAuthorizationFilter(
+        AlertingAuthorizationEntity.Rule,
+        alertingAuthorizationFilterOpts
+      );
+    } catch (error) {
+      this.auditLogger?.log(
+        ruleAuditEvent({
+          action: RuleAuditAction.BULK_EDIT,
+          error,
+        })
+      );
+      throw error;
+    }
+    const { filter: authorizationFilter } = authorizationTuple;
+    const qNodeFilter =
+      (authorizationFilter && filter
+        ? nodeBuilder.and([
+            typeof filter === 'string' ? fromKueryExpression(filter) : filter,
+            authorizationFilter as KueryNode,
+          ])
+        : authorizationFilter) ?? filter;
+
+    const { aggregations, total } = await this.unsecuredSavedObjectsClient.find<
+      RawRule,
+      RuleBulkEditAggregation
+    >({
+      filter: qNodeFilter,
+      page: 1,
+      perPage: 0,
+      type: 'alert',
+      aggs: {
+        alertTypeId: {
+          multi_terms: {
+            terms: [
+              { field: 'alert.attributes.alertTypeId' },
+              { field: 'alert.attributes.consumer' },
+            ],
+          },
+        },
+      },
+    });
+
+    if (total > MAX_RULES_NUMBER_FOR_BULK_EDIT) {
+      throw Error(`More than ${MAX_RULES_NUMBER_FOR_BULK_EDIT} rules matched for bulk edit`);
+    }
+    const buckets = aggregations?.alertTypeId.buckets;
+
+    if (buckets === undefined || buckets.length === 0) {
+      throw Error('Rules not found');
+    }
+
+    await pMap(buckets, async ({ key: [ruleType, consumer] }) => {
+      this.ruleTypeRegistry.ensureRuleTypeEnabled(ruleType);
+
+      await this.authorization.ensureAuthorized({
+        ruleTypeId: ruleType,
+        consumer,
+        operation: WriteOperations.Update,
```

Comment:
we need to create a new `WriteOperations.BulkEdit`, you can use this PR to see what you need to update https://github.com/elastic/kibana/pull/128163